pkg/util/util.go

```diff
@@ -101,3 +101,28 @@ func FilterMap[K comparable, V comparable](m map[K]V, keep func(K, V) bool) map[
 
 	return new
 }
+
+// RequeueError is a specific kind of error that can be returned by operations that need to eventually be requeued following some duration
+type RequeueError struct {
```

Comment:
Not sure this abstraction makes sense. It's the callees responsibility to determine the retry time where you're using it here. I don't think this is needed

Reply:
I thought about that, but I felt like it still makes more sense for something performing validation/verification to be the entity responsible for dictating whether or not a requeue is required. I could have such a method return a ctrl.Result{} with a requeue time in addition to the error but given that we already differentiate user errors from normal errors in validation, I figured this best matches that pattern.